doc/architecture/adr-004.md

```diff
@@ -0,0 +1,273 @@
+  string name = 3;
+  string uri = 4;
+  string data = 5;
+  string sender = 6;
+}
+
+message MsgEditNFTResponse {}
+```
+
+To implement `MsgEditNFT`:
+
+- `id` can be mapped to `id` field of `NFT`
+- `denom_id` can be mapped to `class_id` field of `NFT`
+- `uri` can be mapped to `uri` field of `NFT`
+- Since there is no field for `name` in `NFT`, we'll have to inject name in `data` field of `NFT`
+- `data` can be mapped to `data` field in `NFT`
+
+Once the `NFT` object is prepared, we can call the `UpdateNFT` of `x/nft` module.
+
+#### `MsgTransferNFT`
+
+```protobuf
+service Msg {
+  rpc TransferNFT(MsgTransferNFT) returns (MsgTransferNFTResponse);
+}
+
+message MsgTransferNFT {
+  string id = 1;
+  string denom_id = 2;
+  string sender = 3;
+  string recipient = 4;
+}
+
+message MsgTransferNFTResponse {}
+```
+
+Message to transfer the ownership of NFT is directly provided by Cosmos SDK's NFT module using `MsgSend`. To implement
+`MsgTransferNFT`:
+
+- `id` can be mapped to `id` of `MsgSend`
+- `denom_id` can be mapped to `class_id` of `MsgSend`
+- `sender` can be mapped to `sender` of `MsgSend`
+- `recipient` can be mapped to `receiver` of `MsgSend`
+
+Once `MsgSend` object is created, we can call the `Send` of `x/nft` module.
+
+#### `MsgBurnNFT`
+
+```protobuf
+service Msg {
+  rpc BurnNFT(MsgBurnNFT) returns (MsgBurnNFTResponse);
+}
+
+message MsgBurnNFT {
+  string id = 1;
+  string denom_id = 2;
+  string sender = 3;
+}
+
+message MsgBurnNFTResponse {}
+```
+
+To implement `MsgBurnNFT`, we can directly call the `BurnNFT` of `x/nft` (after performing validations).
+
+### Deprecation Plan
+
+On the release of new wrapper NFT module on Crypto.org chain, all the above messages should be considered as deprecated
+and will be removed in approx. 3 months via an upgrade. After the upgrade, the messages specified by Cosmos SDK's NFT
+module will handle all the operations.
+
+## Status
+
+Draft
+
+## Consequences
+
+### Positive
+
+Cosmos SDK's module will have support from multiple members of the community and will get all the future improvements
+for free (for example, IBC support, inter-module asset support using [ADR-33](https://github.com/cosmos/cosmos-sdk/blob/master/docs/architecture/adr-033-protobuf-inter-module-comm.md))
+
+### Negative
+
+- Our current implementation is slightly different from Cosmos SDK's NFT module which can result in backwards
+  incompatibility. This can be temporarily avoided by creating a wrapper module (which we anyway need to create for 
+  minting and burning NFTs) and adding same messages as the current module along with the sandard message types mentioned
+  [here](https://github.com/cosmos/cosmos-sdk/blob/master/docs/architecture/adr-043-nft-module.md#abstract). But, it is
+  necessary to have a deprecation plan for old message types and transition all the downstream services to new
+  standardized message types.
+- There'll be a lot of duplicate message types until legacy messages are removed.
```

Comment:
the only duplicate message type should be transfer? the others are needed as the wrapper?

Reply:
No. There'll be duplicate messages even for other things (minting, burning, etc.). One will be the current messages and other will be the messages specified in Cosmos SDK's ADR.